gossip/gossip.go

```diff
@@ -218,13 +201,22 @@ func (g *Gossip) GetNodeDescriptor(nodeID proto.NodeID) (proto.NodeDescriptor, e
 // GetNodeDescriptor and internally by getNodeIDAddressLocked.
 func (g *Gossip) getNodeDescriptorLocked(nodeID proto.NodeID) (*proto.NodeDescriptor, error) {
 	nodeIDKey := MakeNodeIDKey(nodeID)
+
+	// We can't use GetInfoProto here because that method grabs the lock.
```

Comment:
Why not make a `getInfoProtoLocked` method and call that from `GetInfoProto` so you don't have to duplicate this unmarshaling code?